client/src/components/UserCard.jsx

```diff
@@ -27,27 +29,55 @@ const UserCard = ({ user }) => {
     itemsByUser[user.id] && setNumOfItems(itemsByUser[user.id].length);
   }, [itemsByUser, user.id]);
 
+  // get link configs from link context
+  useEffect(() => {
+    generateLinkConfigs(user.id);
+  }, [generateLinkConfigs, user.id]);
+
+  // set linkToken and callbacks from configs from link context
+  useEffect(() => {
+    if (linkConfigs.byUser[user.id]) {
+      console.log("i'm inside: ", linkConfigs.byUser[user.id]);
+      setConfig(linkConfigs.byUser[user.id]);
+    }
+  }, [linkConfigs.byUser[user.id]]);
+
   const handleDeleteUser = () => {
     deleteUserById(user.id);
   };
-
+  console.log('did it set it?', config);
   return (
     <div className="box user-card__box">
       <div className="card user-card">
         <div className="user-card__detail">
           <UserDetails user={user} />
         </div>
-        <div className="user-card__button-group">
-          <LinkButton primary userId={user.id}>
-            Link an Item
-          </LinkButton>
-          {!!numOfItems && (
-            <Link className="user-card__link" to={`/user/${user.id}/items`}>
-              {`View ${numOfItems} Linked ${pluralize('Item', numOfItems)}`}
-            </Link>
-          )}
+        <div className="user-card__buttons">
+          <div className="user-card__link__button">
+            {config.token != null && config.onSuccess != null && (
+              <LinkButton userId={user.id} config={config}>
+                Link an Item
+              </LinkButton>
+            )}
+            {!!numOfItems && (
```

Comment:
casting to a boolean using `!!` should not be required since this variable has type `number` right? if it is really required we should A) refactor it so the variable has a type that can be relied on, and B) use `Boolean(numOfItems)` to convert it to a boolean instead of `!!`

```suggestion
            {numOfItems !== 0 && (
```